sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosClientBuilder.java

```diff
@@ -619,6 +621,23 @@ public CosmosClientBuilder readRequestsFallbackEnabled(boolean readRequestsFallb
         return this;
     }
 
+    /**
+     * Sets whether to allow the query plan to be cached when possible, during query execution. Caching query plan
+     * improves the latency/throughput of the query execution when same queries are executed again. It is recommended
+     * to use parameterized queries when trying to cache query plan.
+     * <p>
+     * DEFAULT value is false
+     * </p>
+     *
+     * @param queryPlanCachingEnabled flag to enable query plan cache
```

Comment:
I don't think we should add public surface area for this - why would you ever want to not cache query plan?

Having a system property override to disable it in case we cause some regression etc. is ok - but I would not allow this to be configurable in public API - especially where the best-practice is opt-in

Reply:
This is being enabled as an experimental feature. Based on the feedback we can make it the default behaviour